### PR TITLE
fix: select_v2 asynchronous binding issue (#2368)

### LINKED ISSUE #2368: [Bug Report] select-v2 V-model asynchronous binding data failure
### Element Plus version
1.0.2-beta.53

### OS/Browsers version
chrome

### Vue version
3.1.2

### Reproduction Link
https://codepen.io/EnMcler/pen/KKmKpjG?editors=1111

### Steps to reproduce
see Link

### What is Expected?
see Link

### What is actually happening?
see Link

### PR DESCRIPTION
**Pull request by msidolphin**
Please make sure these boxes are checked before submitting your PR, thank you!

fix #2368

* [x] Make sure you follow Element's contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/select-v2/src/useSelect.ts [hunk 1]
```diff
@@ -571,40 +585,65 @@ const useSelect = (props: ExtractPropTypes<typeof SelectProps>, emit) => {
...
   }
 
-  // in order to track these individually, we need to turn them into refs instead of watching the entire
-  // reactive object which could cause perf penalty when unnecessary field gets changed the watch method will
-  // be invoked.
+  const handleMenuEnter = () => {
+    states.inputValue = states.displayInputValue
+    nextTick(() => {
+      if (~indexRef.value) {
+        scrollToItem(indexRef.value)
```
Comment (JeremyWuuuuu) [inline packages/select-v2/src/useSelect.ts:592]: Fantastic! This is what I wanted to implement.

File: packages/select-v2/src/useSelect.ts [hunk 2]
```diff
@@ -571,40 +585,65 @@ const useSelect = (props: ExtractPropTypes<typeof SelectProps>, emit) => {
...
+    }
+  })
+
+  watch(() => props.modelValue, () => {
+    initStates()
+  })
+
+  watch(() => props.options, () => {
+    initStates()
+  })
```
Comment (JeremyWuuuuu) [inline packages/select-v2/src/useSelect.ts:643]: Should use

```
watch(expression, initStates)
```
?

Since we don't need any parameters here.

**Review by jw-foss — APPROVED** [review]

**Comment by JeremyWuuuuu** [comment]
I left a minor comment, let me know WDYT

File: packages/select-v2/src/useSelect.ts (hunk 2 above)
Comment (msidolphin) [reply packages/select-v2/src/useSelect.ts]: Yeah, i merged them

**Comment by msidolphin** [comment]
> I left a minor comment, let me know WDYT

I already fix.

**Review by jw-foss — APPROVED** [review]

**Pull request merged**